### PR TITLE
No print on wait

### PR DESCRIPTION
**Pull request by tonybaloney**
The `Job.wait_until_completed()` method prints `.` to the console.

Users can fetch the status and the task is blocking. There's no way to disable the print and this is the only method to push data to the console.

### CODE REVIEW

**Comment by vxfield** [comment]
Thanks, @tonybaloney.
I think this was implemented to get some visual progress during interactive scenarios, like in a Jupyter Notebook.

@anpaz, @guenp: What do you think about this scenario? Maybe we can add an optional parameter to disable the print(".") (which will need to be propagated to `get_results`)?

File: azure-quantum/azure/quantum/job.py [hunk 1]
```diff
@@ -61,7 +61,6 @@ def wait_until_completed(self, max_poll_wait_secs=30):
                 f"Waiting for job {self.id},"
                 + f"it is in status '{self.details.status}'"
             )
-            print(".", end="", flush=True)
             time.sleep(poll_wait)
```
Comment (guenp) [inline azure-quantum/azure/quantum/job.py:64]:
```suggestion
            if print_progress:
                print(".", end="", flush=True)
            time.sleep(poll_wait)
```

**Review by guenp — CHANGES_REQUESTED** [review]
I think it's a good idea to disable the print, but to delete it would cause a change in behavior that may confuse some users. Let's add an optional input argument `print_progress` that defaults to `True` to `wait_ujntil_completed`.

**Comment by tonybaloney** [comment]
@guenp changes made

**Review by anpaz — COMMENTED** [review]
lgtm.

**Review by guenp — APPROVED** [review]

**Pull request merged**